### PR TITLE
build: Bump minimum Go version to 1.18

### PR DESCRIPTION
**Pull request by anoopcs9**
_go.sum_ changes are result of running `go mod tidy` after updating **_go_** directive in _go.mod_.

### CODE REVIEW

**Review by synarete — APPROVED** [review]
LGTM

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: "1.17"
+          go-version: "1.18"
```
Comment (phlogistonjohn) [inline .github/workflows/main.yml:16]: IIRC I think setup-go now has symbolic names for the current and current-1 go releases. Once we catch up with go we can maybe switch to the symbolic name.

**Review by phlogistonjohn — APPROVED** [review]
LGTM

File: .github/workflows/main.yml (hunk 1 above)
Comment (phlogistonjohn) [reply .github/workflows/main.yml:16]: i just looked. It's  called "oldstable"  (and  "stable" for latest stable)

**Pull request merged**